### PR TITLE
[HLS-3330] - lambda-sqs context propagation

### PR DESCRIPTION
**Pull request by LironKS**

### CODE REVIEW

File: proxy-libs/helioslambda/lambda.go [hunk 1]
```diff
@@ -22,6 +29,58 @@ type eventBridgeEvent struct {
...
+	}
+	messagingSystem := attribute.KeyValue{
+		Key:   semconv.MessagingSystemKey,
+		Value: attribute.StringValue("aws.sqs"),
+	}
+	messagingPayload := attribute.KeyValue{
+		Key:   "faas.event",
+		Value: attribute.StringValue(record.Body),
+	}
+	updatedCtx, span := tp.Tracer(otellambdaTracerName).Start(recordCtx, "handle_sqs_record", trace.WithAttributes(messageId, messagingSystem, messagingPayload))
```
Comment (nozik) [inline proxy-libs/helioslambda/lambda.go:79]: Can we tell what's the queue name? Is it part of the SQSMEssage interface?

**Review by nozik — CHANGES_REQUESTED** [review]

File: proxy-libs/helioslambda/lambda.go (hunk 1 above)
Comment (LironKS) [reply proxy-libs/helioslambda/lambda.go:79]: is it EventSource?

Comment (nozik) [reply proxy-libs/helioslambda/lambda.go:79]: I guess, please check the implementation in node

File: proxy-libs/helioslambda/lambda.go [hunk 2]
```diff
@@ -22,6 +30,69 @@ type eventBridgeEvent struct {
...
+	result := []string{}
+	for key := range c.messageAttrs {
+		result = append(result, key)
+	}
+
+	return result
+}
+
+func extractQueueName(eventSourceArn string) string {
+	if eventSourceArn != "" {
```
Comment (nozik) [inline proxy-libs/helioslambda/lambda.go:66]: Please do it safely - what happens if ":" are not there

File: proxy-libs/helioslambda/lambda.go [hunk 3]
```diff
@@ -22,6 +30,69 @@ type eventBridgeEvent struct {
...
+	}
+	messagingSystem := attribute.KeyValue{
+		Key:   semconv.MessagingSystemKey,
+		Value: attribute.StringValue("aws.sqs"),
+	}
+	messagingPayload := attribute.KeyValue{
+		Key:   "faas.event",
+		Value: attribute.StringValue(record.Body),
+	}
+	spanName := extractQueueName(record.EventSourceARN)
```
Comment (nozik) [inline proxy-libs/helioslambda/lambda.go:87]: Call it "process {{queueName}}" when we have the queue name

**Review by nozik — CHANGES_REQUESTED** [review]

**Review by nozik — APPROVED** [review]

File: proxy-libs/helioslambda/lambda.go [hunk 4]
```diff
@@ -22,6 +30,72 @@ type eventBridgeEvent struct {
...
+	}
+
+	return result
+}
+
+func extractQueueName(eventSourceArn string) string {
+	if eventSourceArn != "" {
+		queueNameIndex := strings.LastIndex(":", eventSourceArn)
+		if queueNameIndex != -1 {
+			return eventSourceArn[strings.LastIndex(":", eventSourceArn)+1:]
```
Comment (nozik) [inline proxy-libs/helioslambda/lambda.go:69]: Use `queueNameIndex` instead of calling the function again

File: proxy-libs/helioslambda/lambda.go [hunk 5]
```diff
@@ -22,6 +31,72 @@ type eventBridgeEvent struct {
...
+	messagingSystem := attribute.KeyValue{
+		Key:   semconv.MessagingSystemKey,
+		Value: attribute.StringValue("aws.sqs"),
+	}
+	messagingPayload := attribute.KeyValue{
+		Key:   "faas.event",
+		Value: attribute.StringValue(record.Body),
+	}
+	spanName := fmt.Sprintf("process %s", extractQueueName(record.EventSourceARN))
+	if spanName == "" {
```
Comment (nozik) [inline proxy-libs/helioslambda/lambda.go:92]: Span name will never be empty here, you need to check if the queue name is empty

**Review by nozik — APPROVED** [review]

**Pull request merged**